tool/tctl/common/resource_command.go

```diff
@@ -987,13 +1013,22 @@ func (rc *ResourceCommand) Delete(ctx context.Context, client auth.ClientI) (err
 			return trail.FromGRPC(err)
 		}
 		fmt.Printf("login rule %q has been deleted\n", rc.ref.Name)
-
 	case types.KindSAMLIdPServiceProvider:
 		if err := client.DeleteSAMLIdPServiceProvider(ctx, rc.ref.Name); err != nil {
 			return trace.Wrap(err)
 		}
 		fmt.Printf("SAML IdP service provider %q has been deleted\n", rc.ref.Name)
+	case types.KindDevice:
+		remote := client.DevicesClient()
+		device, err := findDeviceByIDOrTag(ctx, remote, rc.ref.Name)
+		if err != nil {
+			return trace.Wrap(err)
+		}
 
+		if _, err := remote.DeleteDevice(ctx, &devicepb.DeleteDeviceRequest{DeviceId: device.Id}); err != nil {
```

Comment:
Suggestion: Line breaks for readability:

```suggestion
		if _, err := remote.DeleteDevice(ctx, &devicepb.DeleteDeviceRequest{
			DeviceId: device.Id,
		}); err != nil {
```